File: Data Structures & Algorithms/non-overlapping-intervals/submission-0.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end(), [&](vector<int> a, vector<int> b) {
            return a[1] < b[1];
        });

        int n = intervals.size();
        vector<int> f(n, 0);
        f[0] = 1;
        for (int i = 1; i < n; i++) {
            int l = 0;
            int r = i - 1;
            int k = -1;
            while (l <= r) {
                int mid = (r - l) / 2 + l;
                if (intervals[mid][1] <= intervals[i][0]) {
                    k = mid;
                    l = mid + 1;
                } else {
                    r = mid - 1;
                }
            }

            f[i] = f[i - 1];
            if (k != -1) {
                f[i] = max(f[i], f[k] + 1);
            }
        }
        return n - f[n - 1];
    }
};

File: Data Structures & Algorithms/task-scheduling/submission-0.py
class Solution:
    def leastInterval(self, tasks: List[str], n: int) -> int:
        freq = [0] * 26
        for task in tasks:
            freq[ord(task) - ord('A')] += 1

        freq.sort()
        maxf = freq[25]
        idles = (maxf - 1) * n

        for i in range(24,-1,-1):
            idles -= min(freq[i], maxf - 1)

        return max(idles, 0) + len(tasks)
        